clients/da-vinci-client/src/main/java/com/linkedin/davinci/blob/BlobSnapshotManager.java

```diff
@@ -0,0 +1,116 @@
+package com.linkedin.davinci.blob;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.store.rocksdb.RocksDBUtils;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.rocksdb.Checkpoint;
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksDBException;
+
+
+public class BlobSnapshotManager {
+  private final String basePath;
+
+  @VisibleForTesting
+  protected Checkpoint createCheckpoint(RocksDB rocksDB) {
+    return Checkpoint.create(rocksDB);
+  }
+
+  private static final Logger LOGGER = LogManager.getLogger(BlobSnapshotManager.class);
+  HashMap<String, Map<Integer, AtomicLong>> concurrentSnapshotUsers;
+  HashMap<String, HashMap<Integer, Long>> snapShotTimestamps;
+  Long snapshotRetentionTime;
+
+  // Initialized the snapshot manger for this specific dbDir
+  public BlobSnapshotManager(String basePath, long snapshotRetentionTime) {
+    this.basePath = basePath;
+    this.snapshotRetentionTime = snapshotRetentionTime;
+    this.snapShotTimestamps = new HashMap<>();
+    this.concurrentSnapshotUsers = new HashMap<>();
+  }
+
+  // Checks if the current snapshot of the partition is stale
+  private boolean isSnapshotStale(String topicName, int partitionId) {
+    if (snapShotTimestamps.get(topicName) == null || snapShotTimestamps.get(topicName).get(partitionId) == null) {
+      return true;
+    }
+    long snapshotTimestamp = snapShotTimestamps.get(topicName).get(partitionId);
+    return System.currentTimeMillis() - snapshotTimestamp > this.snapshotRetentionTime;
```

Comment:
```suggestion
    long snapshotTimestamp = getSnapshotTimestamp(this.snapshotTimestamps);
    if (snapshotTimestamp == null) {
      return true;
    }
    return System.currentTimeMillis() - snapshotTimestamp > this.snapshotRetentionTime;
```
It feels like you're dancing around the fact that these two `HashMap` data structures `concurrentSnapshotUsers` and `snapShotTimestamps` are 2-dimensional, and individually doing error handling at every juncture.
I feel like this is the best example of when helper methods would be effective. Something like `getSnapshotTimestamp(this.snapshotTimestamps)`, which would either return `null` or throw an exception which would be handled by this function? I feel that this would help this entire file out a lot.

Reply:
```suggestion
    try {
      long snapshotTimestamp = snapShotTimestamps.get(topicName).get(partitionId);
      return System.currentTimeMillis() - snapshotTimestamp > this.snapshotRetentionTime;
    } except (NoSuchElementException e) {
      return true;
    }
```

Reply:
Implemented the simplification you suggested